### ISSUE TITLE
mat1 and mat2 shapes cannot be multiplied (2x1024 and 768x320)

### ISSUE THREAD

**Issue by zymox** [MrForExample/ComfyUI-AnimateAnyone-Evolved#17]
Trying to run the example workflow (with the provided example video + image), I get an error with the sampler:

```
Error occurred when executing [ComfyUI-3D] Animate Anyone Sampler:

mat1 and mat2 shapes cannot be multiplied (2x1024 and 768x320)

File "E:\ComfyUI\execution.py", line 155, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
File "E:\ComfyUI\execution.py", line 85, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
File "E:\ComfyUI\execution.py", line 78, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
File "E:\ComfyUI\custom_nodes\ComfyUI-AnimateAnyone-Evolved\nodes.py", line 152, in animate_anyone
samples = diffuser(
File "C:\Dev\Python3.10\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
return func(*args, **kwargs)
File "E:\ComfyUI\custom_nodes\ComfyUI-AnimateAnyone-Evolved\src\models\main_diffuser.py", line 440, in __call__
latents = self.denoise_loop(
File "E:\ComfyUI\custom_nodes\ComfyUI-AnimateAnyone-Evolved\src\models\main_diffuser.py", line 315, in denoise_loop
self.reference_unet(
File "C:\Dev\Python3.10\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\ComfyUI\custom_nodes\ComfyUI-AnimateAnyone-Evolved\src\models\unet_2d_condition.py", line 1197, in forward
sample, res_samples = downsample_block(
File "C:\Dev\Python3.10\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\ComfyUI\custom_nodes\ComfyUI-AnimateAnyone-Evolved\src\models\unet_2d_blocks.py", line 657, in forward
hidden_states, ref_feature = attn(
File "C:\Dev\Python3.10\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\ComfyUI\custom_nodes\ComfyUI-AnimateAnyone-Evolved\src\models\transformer_2d.py", line 357, in forward
hidden_states = block(
File "C:\Dev\Python3.10\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "E:\ComfyUI\custom_nodes\ComfyUI-AnimateAnyone-Evolved\src\models\mutual_self_attention.py", line 241, in hacked_basic_transformer_inner_forward
attn_output = self.attn2(
File "C:\Dev\Python3.10\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "C:\Dev\Python3.10\lib\site-packages\diffusers\models\attention_processor.py", line 527, in forward
return self.processor(
File "C:\Dev\Python3.10\lib\site-packages\diffusers\models\attention_processor.py", line 1246, in __call__
key = attn.to_k(encoder_hidden_states, *args)
File "C:\Dev\Python3.10\lib\site-packages\torch\nn\modules\module.py", line 1501, in _call_impl
return forward_call(*args, **kwargs)
File "C:\Dev\Python3.10\lib\site-packages\diffusers\models\lora.py", line 430, in forward
out = super().forward(hidden_states)
File "C:\Dev\Python3.10\lib\site-packages\torch\nn\modules\linear.py", line 114, in forward
return F.linear(input, self.weight, self.bias)
``` 

No idea how to approach this ... any help?

**Comment by zymox** [comment]
Figured it out. 

I was using a different CLIP Vision model, changed it to the one mentioned in the install instructions (https://huggingface.co/lambdalabs/sd-image-variations-diffusers/tree/main/image_encoder) and it works.

**Comment by nothingness6** [comment]
> Figured it out.
> 
> I was using a different CLIP Vision model, changed it to the one mentioned in the install instructions (https://huggingface.co/lambdalabs/sd-image-variations-diffusers/tree/main/image_encoder) and it works.

I also faced the same issue, but it doesn't solve my issue. Where should I put it into?

**Comment by DougPP** [comment]
> Figured it out.
> 
> I was using a different CLIP Vision model, changed it to the one mentioned in the install instructions (https://huggingface.co/lambdalabs/sd-image-variations-diffusers/tree/main/image_encoder) and it works.

I am facing the same error "mat1 and mat2 shapes cannot be multiplied (2x1024 and 768x320)".
I followed your instruction & made sure the correct model was downloaded but still receive the same error message.

Anyone else able to resolve this?

**Comment by Potts2k8** [comment]
> Figured it out.
> 
> I was using a different CLIP Vision model, changed it to the one mentioned in the install instructions (https://huggingface.co/lambdalabs/sd-image-variations-diffusers/tree/main/image_encoder) and it works.

Noob here. Where do we even put this?

**Comment by changer1666** [comment]
Because you're using SDXL's checkpoint model, controlnet uses SD1.5's

**Comment by changer1666** [comment]
Because you're using SDXL's checkpoint model, but,controlnet uses SD1.5's

**Comment by despallier** [comment]
still playing around with comfyui. I get an error :executing KSampler:

'ModuleList' object has no attribute '1'

File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\execution.py", line 151, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\execution.py", line 81, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\execution.py", line 74, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1373, in sample
return common_ksampler(model, seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise=denoise)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\nodes.py", line 1343, in common_ksampler
samples = comfy.sample.sample(model, noise, steps, cfg, sampler_name, scheduler, positive, negative, latent_image,
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\comfy\sample.py", line 43, in sample
samples = sampler.sample(noise, positive, negative, cfg=cfg, latent_image=latent_image, start_step=start_step, last_step=last_step, force_full_denoise=force_full_denoise, denoise_mask=noise_mask, sigmas=sigmas, callback=callback, disable_pbar=disable_pbar, seed=seed)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 801, in sample
return sample(self.model, noise, positive, negative, cfg, self.device, sampler, sigmas, self.model_options, latent_image=latent_image, denoise_mask=denoise_mask, callback=callback, disable_pbar=disable_pbar, seed=seed)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 703, in sample
return cfg_guider.sample(noise, latent_image, sampler, sigmas, denoise_mask, callback, disable_pbar, seed)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 690, in sample
output = self.inner_sample(noise, latent_image, device, sampler, sigmas, denoise_mask, callback, disable_pbar, seed)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 665, in inner_sample
self.conds = process_conds(self.inner_model, noise, self.conds, device, latent_image, denoise_mask, seed)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 627, in process_conds
pre_run_control(model, conds[k])
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\comfy\samplers.py", line 475, in pre_run_control
x['control'].pre_run(model, percent_to_timestep_function)
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\comfy\controlnet.py", line 316, in pre_run
comfy.utils.set_attr_param(self.control_model, k, self.control_weights[k].to(dtype).to(comfy.model_management.get_torch_device()))
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\comfy\utils.py", line 455, in set_attr_param
return set_attr(obj, attr, torch.nn.Parameter(value, requires_grad=False))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\ComfyUI\comfy\utils.py", line 449, in set_attr
obj = getattr(obj, name)
^^^^^^^^^^^^^^^^^^
File "C:\comfyui\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\modules\module.py", line 1709, in __getattr__
raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")

**Comment by YuanJie2001** [comment]
What should I do?

![Image](https://github.com/user-attachments/assets/043c6441-52b9-4a48-990a-659c763591db)
